File: docs/code-examples/point3d_random_v2.py
"""Log some random points with color and radii."""
import rerun as rr
import rerun.experimental as rr2
from numpy.random import default_rng

rr.init("rerun-example-points3d_random", spawn=True)
rng = default_rng(12345)

positions = rng.uniform(-5, 5, size=[10, 3])
colors = rng.uniform(0, 255, size=[10, 3])
radii = rng.uniform(0, 1, size=[10])

rr2.log("random", rr2.Points3D(positions, colors=colors, radii=radii))

File: docs/code-examples/point3d_random_v2.rs
//! Log some random points with color and radii.

use rand::distributions::Uniform;
use rand::Rng;
use rerun::{archetypes::Points3D, components::Color, MsgSender, RecordingStreamBuilder};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec_stream, storage) =
        RecordingStreamBuilder::new("rerun-example-points3d_random").memory()?;

    let mut rng = rand::thread_rng();
    let dist = Uniform::new(-5., 5.);

    MsgSender::from_archetype(
        "random",
        &Points3D::new((0..10).map(|_| (rng.sample(dist), rng.sample(dist), rng.sample(dist))))
            .with_colors((0..10).map(|_| Color::from_rgb(rng.gen(), rng.gen(), rng.gen())))
            .with_radii((0..10).map(|_| rng.gen::<f32>())),
    )?
    .send(&rec_stream)?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}
